### ISSUE TITLE
Overwatch Profile does not work

### ISSUE THREAD

**Issue by mohsin-mulla** [SpoinkyNL/Artemis#136]
I just discovered this awesome app and I really like what you've done so far! :+1: 

However, I can't seem to get the Overwatch profile to work. Whenever I launch the game, all of the keyboard lights go off until I exit the game. This is using a Corsair Strafe RGB keyboard.

Hope this can be fixed soon, keep up the great work!

**Comment by SpoinkyNL** [comment]
Hello, Mohsin

Thanks for trying Artemis :)
Could you double check that the folder Artemis detected for Overwatch is
correct, and that it contains a file called RzChromaSDK64

Op 9 aug. 2016 04:12 schreef "Mohsin Mulla" <notifications@github.com>:

**Comment by mohsin-mulla** [comment]
Hello!

Yup, here's a screenshot showing the directory structure located under `C:\Program Files (x86)\Overwatch:`
![explorer_2016-08-09_19-21-32](https://cloud.githubusercontent.com/assets/7242533/17528260/91b2f8a2-5e66-11e6-8b17-d73b36e89548.png)

**Comment by SpoinkyNL** [comment]
Alright that seems to be fine.

If you go to settings and open the debug window, does it shows what
Overwatch is outputting to Razer, does this show anything?

It should be orange in the main menu and change colors in the character
selection screen.

If this doesn't show up, try running both Artemis and Overwatch as
administrator

Op 9 aug. 2016 8:22 p.m. schreef "Mohsin Mulla" <notifications@github.com>:

**Comment by mohsin-mulla** [comment]
The debugger doesn't seem to show anything when running Overwatch, and the keyboard lights remain off no matter what screen I am on in game. I've pressed the brightness button on the keyboard too in case the lights had been switched off but this was not the case.

I've run both Artemis and Overwatch as administrator and this hasn't made a difference either

**Comment by SpoinkyNL** [comment]
Hmm ok very strange, what virus scanner are you using? Perhaps you could try temporarily disabling it and see if that helps 

If not I'll need to investigate further, it's always hard solving something that doesn't occur on my own PC, but I'm positive we'll figure it out

**Comment by mohsin-mulla** [comment]
I'm using the one that's included with Windows 10. I've tried running it with the AV disabled but no luck.

Could you tell me where the log files are stored? Maybe I could delete all the logs, change logging to trace and then run Overwatch?

**Comment by SpoinkyNL** [comment]
You can find the log files over at Documents > Artemis > logs

I don't expect there to be much of use in there though. While working on the new update I did notice that the named pipe server (which is used by Artemis to communicate with the game) doesn't give an error message when something goes wrong.

I changed that, so now when it fails it'll give an error message instead of pretending nothing is happening.

I'm going to upload the release in a few hours so if you could try it with that (it won't fix your issue but it'll hopefully help me better understand what's wrong)

**Comment by IgorP25** [comment]
I have the same issue on a Corsair K95. Running Overwatch results in no keyboard lighting at all. The profile itself works fine when previewed within Artemis. That is, if I duplicate the default Overwatch profile and click each layer the keyboard lights up as expected. As soon as I run Overwatch the keyboard goes black.

I also get no output at all in the debug window while Overwatch is running.

All of this was done running the latest 1.2.3.0 release.

I have attached my logs in case it helps.
[2016-08-15.txt](https://github.com/SpoinkyNL/Artemis/files/417228/2016-08-15.txt)

**Comment by SpoinkyNL** [comment]
@IgorP25 In your logs I don't see anything about Artemis placing the Overwatch DLL. Could you shut down Overwatch and make sure the game directory is correctly selected?

**Comment by mohsin-mulla** [comment]
Hey, I've installed the latest release and attached the log files for you to have a look at!
[2016-08-15.txt](https://github.com/SpoinkyNL/Artemis/files/417371/2016-08-15.txt)

**Comment by SpoinkyNL** [comment]
Hmm looking at that log everything seems fine at the Artemis side.. I had hoped to find some error related to Overwatch in there but it seems to setup the named pipe just fine. 

There is one error in there but that's not what's causing this, and already fixed. 

If you don't mind I'd like to create a special version of the Overwatch dll with logging for you to try, so I can hopefully tell what's going wrong. Let me know if you're willing to give that a go

**Comment by mohsin-mulla** [comment]
Sure, I'll give it a go

**Comment by IgorP25** [comment]
@SpoinkyNL Not sure what happened there. It was obviously detecting when Overwatch runs because the keyboard goes black. I reselected the Overwatch folder however and the relevant line seems to be in the logs now `|DEBUG|Artemis.Utilities.DataReaders.DllManager|Successfully placed Razer DLL in C:\Program Files (x86)\Overwatch `

However the result is still the same. Keyboard still black. Perhaps it's an issue with Overwatch itself?

Here is a fresh log since I reselected the folder.

[2016-08-16.txt](https://github.com/SpoinkyNL/Artemis/files/418435/2016-08-16.txt)

**Comment by SpoinkyNL** [comment]
Alright, could you both try it with this file: https://www.dropbox.com/s/p6ossma5uj6ddt2/RzChromaSDK64.dll?dl=0

To use it please follow these steps, in this order:

1. Start up Artemis
2. Copy the RzChromaSDK64.dll file to the Overwatch directory
3. Startup Overwatch
4. Go to Training > Practice Range
5. In the select screen, browse through the heroes
6. Shut down Overwatch
7. If everything went well there's a file called log.txt in your Overwatch folder, please post the contents here

Thanks guys!

**Comment by IgorP25** [comment]
It didn't go well for me unfortunately. Performed the above steps but no log.txt file appeared in the Overwatch folder. I also tried running Overwatch as admin but this didn't produce the log file either. Any ideas?

**Comment by SpoinkyNL** [comment]
Hmm that's really strange, like it's ignoring the DLL altogether o.O

Could you try removing it from the Overwatch folder (while Artemis is running, but Overwatch is not) and placing it in C:\Windows\System32?

**Comment by IgorP25** [comment]
No difference. No complaints from Overwatch even if I remove the file completely from both locations either. So I guess it never loads or it has an alternate somewhere. Although my understanding is DLLs in the program dir override any others yes?

**Comment by SpoinkyNL** [comment]
Yes, that's how I've always understood it as well. Do you have or ever had Razer Synapse (the software that comes with Razer gear) installed?

**Comment by IgorP25** [comment]
Yeah I do. Mouse is a Razer Mamba 2012 (does not change colour as part of chroma apps) so I'm running Synapse for that.

Do you think that's causing issues?

**Comment by SpoinkyNL** [comment]
Maybe, could you try uninstalling it, and rebooting, for science?

**Comment by IgorP25** [comment]
No change. You win this round science!

**Comment by SpoinkyNL** [comment]
Hmm too bad, I'm really only able to do guesswork so could you try installing https://www.microsoft.com/en-us/download/details.aspx?id=48145 Visual Studio C++ 2015 Redistributable 64 bits? And try again (Using the DLL with logging)

**Comment by IgorP25** [comment]
Ok so here's where things get weird. Installing the above also seemingly had no effect, still no log file.
Just to cover all the bases I ran "verify game files" in battle.net. After this I restarted Artemis and ran Overwatch as admin, and without replacing the DLL. This time they lighting functions fired up. I went into training mode, changed heroes, all working as expected. To verify I started Overwatch a few more times without admin mode, restarted the computer and tried again, the lighting worked every time.

At this point out of curiosity I tried changing out the DLL to the logging version. I discovered you need to go to the properties of the DLL file and unblock it because it detects it as being "from another system" and therefore "dangerous" so it is blocked by default. I fired up Overwatch, but again, no lighting and then no log file.

So in short I'm pretty sure a simple verification of the game files fixed my issue, but I'm not sure what caused the problem, and I never got the log DLL working.

I'm not sure if this will work for mohsin-mulla but it's worth a try.

Thanks for the help by the way, I know it's a pain to diagnose stuff like this.

**Comment by SpoinkyNL** [comment]
Oh wow I didn't think of that yet, good catch! Thanks for the detailed report perhaps it'll be some use to @mohsin-mulla too :)

**Comment by mohsin-mulla** [comment]
Okay, so I replaced the DLL file with the one you provided, unblocked it as per @IgorP25 instructions and then started Overwatch (both non-admin and admin). No log file was created and the lights were still not working.

Next, I installed the Visual Studio C++ Redistributable you linked to and started Overwatch (both non-admin and admin). Again, no log file and lights just like before.

Then, I closed Razer Synapse in case it was interfering with anything and voila! It started to work. From here, I thought it might be an issue with Synapse, but opening Synapse and running Overwatch... the lights still work.

I have _no idea_ what fixed it, either the combination of installing the C++ Redistributable or closing Synapse did it... or just pure luck. 

So I guess a possible solution for anyone else having issues is, install the Visual Studio C++ Redistributable linked above, and close Synapse and see if that helps.

**Comment by SpoinkyNL** [comment]
Alright this is great news, I'll put it in the FAQ so people who are running into trouble can also try this out. Thanks to both of you for reporting back in such detail!

I'll leave this open untill I get around to updating the FAQ

**Comment by mohsin-mulla** [comment]
No worries, thank _you_ for the help and for working on this :)

**Comment by sifunex** [comment]
I totally had this happen, I needed to run as overwatch with administrator... then it started to work properly.

**Comment by lolitsjohn** [comment]
Having the same issue. Logitech G810 here. Did we figure out exactly what was causing the issue? I tried installing that C++ Redistributible, launch as admin, restarting. Still nothing =(.
